accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+
+  The core of the CaaS Platform is master and worker nodes. The Management Node, instructed by SUSE Manager, will take care of bootstrapping master and worker nodes.
+
+  The critical part of every master and worker node is handling updates of the underlying Kubernetes-related packages (`kubernetes-kubeadm kubernetes-kubelet kubernetes-client cri-o cni-plugins`): these updates must be handled with `skuba` from the Management Node.
+
+  [`skuba-update` will take care of updating the Base Operating System](https://susedoc.github.io/doc-caasp/beta/caasp-admin/single-html/#_base_os): this means we cannot register any master or worker node with SUSE Manager (otherwise an end-user could apply updates to the machine).
+  In case the machines are registered to SUSE Manager, the machines will be de-registered during the bootstrapping.
+  
+  ###### Bootstrapping master and worker nodes
+
+  In SUSE Manager under "CaaS Platform > Deployment > cluster name" there will be a list of machines that:
+  - are not part of any other CaaS Platform cluster
+  - are not already bootstrapped into the current cluster
+  - satisfy the CaaS Platform node requirements: those machines are either bootstrapped via the AutoYaST profile or are already registered with SUSE Manager and satisfy the node requirements (checked via grains).
+
+  The user will then have the option to bootstrap the selected machine as the first master (or subsequently, as an additional master or worker). Upon triggering of this event, the SUSE Manager will:
+
+  - operate on the to-be master or worker node:
+    - Copy the SSH key of the Management Node for passwordless login in the node. In case of existing machines, the SSH file must be copied (e.g. via Salt), whereas the AutoYaST case will be already covered during provisioning.
+    - Assign the CaaS Platform channels assigned via Salt (SUSE CaaS Platform Extension to SUSE Linux Enterprise 15 and SUSE Containers Module 15 SP1 x86_64), whereas the AutoYaST case will be already covered during provisioning.
+    - De-register the machine from SUSE Manager (if it is registered)
+
+  - operate on the Management Node:
+    - Salt will invoke `skuba` (using `cmd.run`) to bootstrap the first master or make the selected node join the cluster. The user must specify which user to use for passwordless login (in this case `sles`) and whether to use `sudo` or not.
+
+    Example:
+
+    ```
+    bootstrap_first_master:
+      cmd.run:
+        - name: skuba node bootstrap --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_additional_master:
+      cmd.run:
+        - name: skuba node join --role master --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_worker:
+      cmd.run:
+        - name: skuba node join --role worker --user sles --sudo --target <node IP> <node name>
+    ```
+
+    The output of the `skuba` command must be parsed and shown to the user in SUSE Manager in case of errors.
```

Comment:
I suggest to use the retcode instead of parsing, and maybe show the raw output. If you have a requirement in which you want to show more meaningful information we should sync on how this information could be provided to SUMA in a machine readable way instead of parsing human output that is very likely to change.

Reply:
How about exporting a JSON output with `retcode` and a possible `errorMessage`?

Reply:
In the meanwhile, I added that we are going to check for the return code and in case of errors show the raw output.